pkg/remotecmd/interface.go

```diff
@@ -0,0 +1,66 @@
+package remotecmd
+
+import (
+	devfilev1 "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+
+	"github.com/redhat-developer/odo/pkg/kclient"
+)
+
+// RemoteProcessStatus is an enum type for representing process statuses.
+type RemoteProcessStatus int
+
+const (
+	// Unknown represents a process for which the status cannot be determined reliably or is not handled yet by us.
+	Unknown RemoteProcessStatus = iota + 1
+
+	// Starting represents a process that is just about to start.
+	Starting
+
+	// Stopped represents a process stopped.
+	Stopped
+
+	// Running represents a running process.
+	Running
+)
```

Comment:
> This bit is interesting, I haven't seen anything like it before! 
> But I am wondering, if changing the order of these constants, thereby changing their values, will have any effect to their usages.
>
The values do not matter here; we refer to the constants by name rather than by value. So changing the order (or even inserting new statuses in the list) will have no effect.
More details in the coding conventions doc: https://github.com/redhat-developer/odo/wiki/Dev:-Coding-Conventions#start-enums-at-one